silkworm/core/execution/evm.cpp

```diff
@@ -149,7 +150,9 @@ evmc::Result EVM::create(const evmc_message& message) noexcept {
         state_.set_nonce(contract_addr, 1);
     }
 
-    state_.subtract_from_balance(message.sender, value);
+    if (have >= value) {
```

Comment:
I would change this condition to:
```c++
if (!gas_bailout_ || have >= value)
```
in order to be more explicit and maybe also exploit short-circuiting.

Reply:
ok